core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java

```diff
@@ -506,8 +508,13 @@ public void write(final String name, final byte[] data) {
                 // Build the config
 
                 final ResultHandle config = carc.checkCast(carc.invokeVirtualMethod(SRCB_BUILD, builder), SmallRyeConfig.class);
-                final ResultHandle providerResolver = carc.newInstance(SCPR_CONSTRUCT, config);
-                carc.invokeStaticMethod(CPR_SET_INSTANCE, providerResolver);
+
+                // ConfigProviderResolver.setInstance(new SimpleConfigurationProviderResolver())
+                carc.invokeStaticMethod(CPR_SET_INSTANCE,
```

Comment:
There doesn't seem to be a good reason to change this?

Reply:
You do mention your reasons for using a static field, but it's not really clear why you would change the constructor from one which accepts the configuration to one which does not?

Reply:
The constructor with param is useless because we always need to set the config by means of `registerConfig()`.